### PR TITLE
add automatic db deployment to test, uat. add keycloak sync to uat. update action definitions to allow for env specific secrets for keycloak sync.

### PR DESCRIPTION
**Pull request by devinleighsmith**

### CODE REVIEW

**Comment by devinleighsmith** [comment]
secret creation covered by: https://jira.th.gov.bc.ca/browse/PSP-7200

File: .github/workflows/ci-cd-pims-dev.yml [hunk 1]
```diff
@@ -123,7 +123,7 @@ jobs:
       - name: call scripts to upgrade database
         shell: bash
         run: |
-          oc process -f ./openshift/4.0/templates/jobs/db-deploy.yaml -p SERVER_NAME=sqldevtst.th.gov.bc.ca -p DB_NAME=PIMS_DEV -p NAMESPACE=3cd915-dev | oc create -f - | grep -oP "(?<=job\.batch/)[^\s]*" | (read JOB_NAME; oc wait --for=condition=complete job/$JOB_NAME --timeout=120s)
+          oc process -f ./openshift/4.0/templates/jobs/db-deploy.yaml -p DB_SECRET_NAME=pims-database GIT_BRANCH=dev SERVER_NAME=sqldevtst.th.gov.bc.ca -p DB_NAME=PIMS_DEV -p NAMESPACE=3cd915-dev | oc create -f - | grep -oP "(?<=job\.batch/)[^\s]*" | (read JOB_NAME; oc wait --for=condition=complete job/$JOB_NAME --timeout=120s)
```
Comment (asanchezr) [inline .github/workflows/ci-cd-pims-dev.yml:126]: did you test this on OCP? According to **OC** documentation, template parameters need to be listed **each** with `-p` but I see this `oc process ... -p DB_SECRET_NAME=... [-p needed here?] GIT_BRANCH=dev ...`

I think you need to add -p in front of GIT_BRANCH and SERVER_NAME parameters - right? or does it work without it?

File: .github/workflows/retag-dev-to-test.yml [hunk 2]
```diff
@@ -64,9 +64,33 @@ jobs:
...
+        with:
+          openshift_server_url: ${{ env.OPENSHIFT_SERVER }}
+          openshift_token: ${{ env.OPENSHIFT_TOKEN }}
+          insecure_skip_tls_verify: true
+          namespace: 3cd915-dev
+      - name: call scripts to upgrade database
+        shell: bash
+        run: |
+          oc process -f ./openshift/4.0/templates/jobs/db-deploy.yaml -p DB_SECRET_NAME=pims-database-test GIT_BRANCH=dev SERVER_NAME=sqldevtst.th.gov.bc.ca -p DB_NAME=PIMS_TST -p NAMESPACE=3cd915-dev | oc create -f - | grep -oP "(?<=job\.batch/)[^\s]*" | (read JOB_NAME; oc wait --for=condition=complete job/$JOB_NAME --timeout=120s)
+
```
Comment (asanchezr) [inline .github/workflows/retag-dev-to-test.yml:89]: Same comment re: `-p` applies here

File: .github/workflows/retag-test-to-uat.yml [hunk 3]
```diff
@@ -59,11 +61,59 @@ jobs:
...
+        uses: redhat-actions/oc-login@v1
+        with:
+          openshift_server_url: ${{ env.OPENSHIFT_SERVER }}
+          openshift_token: ${{ env.OPENSHIFT_TOKEN }}
+          insecure_skip_tls_verify: true
+          namespace: 3cd915-test
+      - name: call scripts to upgrade database
+        shell: bash
+        run: |
+          oc process -f ./openshift/4.0/templates/jobs/db-deploy.yaml -p DB_SECRET_NAME=pims-database GIT_BRANCH=test SERVER_NAME=sqlprd.th.gov.bc.ca -p DB_NAME=PIMS_UAT -p NAMESPACE=3cd915-test | oc create -f - | grep -oP "(?<=job\.batch/)[^\s]*" | (read JOB_NAME; oc wait --for=condition=complete job/$JOB_NAME --timeout=120s)
```
Comment (asanchezr) [inline .github/workflows/retag-test-to-uat.yml:86]: please check usage of `-p` here

File: .github/workflows/retag-test-to-uat.yml [hunk 4]
```diff
@@ -59,11 +61,59 @@ jobs:
...
+        uses: actions/checkout@v3
+
+      - name: Setup .NET 6
+        uses: actions/setup-dotnet@v2
+        with:
+          dotnet-version: "6.0.x"
+
+      - name: Install dependencies for keycloak sync
+        run: dotnet restore
+        working-directory: ${{env.sync-directory}}
```
Comment (asanchezr) [inline .github/workflows/retag-test-to-uat.yml:103]: The env-var **sync-directory: ./tools/keycloak/sync** is not present in this file like to others above. Please add it or kc-sync will fail in UAT

**Review by asanchezr — APPROVED** [review]
See comments

**Pull request merged**